### ISSUE TITLE
Parser meta lookups don't work correctly when using DELETE .. FROM with aliased tables

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14353]
This test fails:

```java
String sql =
    """
    DELETE a
    FROM a x, b y
    WHERE x.c1 = y.c1
    """;

Meta meta = dsl.meta("""
    create table a (c1 int);
    create table b (c1 int);
    """);
dsl.configuration().set((MetaProvider) () -> meta);
dsl.settings().setParseWithMetaLookups(ParseWithMetaLookups.THROW_ON_FAILURE);
println(dsl.parser().parseQuery(sql));
```

The error is:

```
org.jooq.impl.ParserException: Unknown field identifier: [3:7] DELETE a
FROM a x, b y
WHERE [*]x.c1 = y.c1

	at org.jooq.impl.DefaultParseContext.exception(ParserImpl.java:14374)
	at org.jooq.impl.DefaultParseContext$ParseScope.unknownField(ParserImpl.java:14743)
	at org.jooq.impl.DefaultParseContext$ParseScope.scopeEnd(ParserImpl.java:14702)
	at org.jooq.impl.DefaultParseContext.parseQuery(ParserImpl.java:1262)
	at org.jooq.impl.DefaultParseContext.lambda$2(ParserImpl.java:947)
	at org.jooq.impl.DefaultParseContext.wrap(ParserImpl.java:14567)
	at org.jooq.impl.DefaultParseContext.parseQuery0(ParserImpl.java:947)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:764)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:759)
	at org.jooq.impl.ParserMetaTest.x(ParserMetaTest.java:1420)
```

Note that this SQL also doesn't parse:

```java
String sql =
    """
    DELETE a
    FROM a, b
    WHERE a.c1 = b.c1
    """;
```

The error is now:

```
org.jooq.impl.ParserException: Unknown field identifier: [3:14] DELETE a
FROM a, b
WHERE a.c1 = [*]b.c1

	at org.jooq.impl.DefaultParseContext.exception(ParserImpl.java:14374)
	at org.jooq.impl.DefaultParseContext$ParseScope.unknownField(ParserImpl.java:14743)
	at org.jooq.impl.DefaultParseContext$ParseScope.scopeEnd(ParserImpl.java:14702)
	at org.jooq.impl.DefaultParseContext.parseQuery(ParserImpl.java:1262)
	at org.jooq.impl.DefaultParseContext.lambda$2(ParserImpl.java:947)
	at org.jooq.impl.DefaultParseContext.wrap(ParserImpl.java:14567)
	at org.jooq.impl.DefaultParseContext.parseQuery0(ParserImpl.java:947)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:764)
	at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:759)
	at org.jooq.impl.ParserMetaTest.x(ParserMetaTest.java:1420)
```

**Comment by lukaseder** [comment]
`UPDATE .. FROM` is also affected

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14354), and 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14355)
